common/src/main/java/com/wynntils/screens/widgets/SearchWidget.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.screens.settings.WynntilsSettingsScreen;
+import java.util.Objects;
+import java.util.function.Consumer;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.resources.sounds.SimpleSoundInstance;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.sounds.SoundEvents;
+
+public class SearchWidget extends TextInputBoxWidget {
+    private final Component DEFAULT_TEXT = new TranslatableComponent("screens.wynntils.searchWidget.defaultSearchText");
+
+    public SearchWidget(
+            int x,
+            int y,
+            int width,
+            int height,
+            Consumer<String> onUpdateConsumer,
+            WynntilsSettingsScreen settingsScreen) {
+        super(x, y, width, height, new TextComponent("Search..."), onUpdateConsumer, settingsScreen);
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        this.renderBg(poseStack, McUtils.mc(), mouseX, mouseY);
```

Comment:
Maybe rename it `renderBackground` to match other places?

Reply:
This is what the superclass method is called.

Reply:
And you can't use that? What's the difference by the super's backgorund render method, and your?

Reply:
I mean superclass's method is called renderBg, and I would rather override that than make a new method.

Reply:
Ah, ok.